### PR TITLE
Shared homes

### LINKED ISSUE #45: shared homes
shared homes with other players. so you type 
`!home share Joshyray`
to accept
`!home share accept`

Give a name to the home by 
`!home share Joshyray village`
accepting players can name home by
`!home share accept village`

AC:
* Names are specific to the player on the command they input to  make the home
* Players can only share one home per "couple" per two people
* Players can delete their side of the home but cannot add a new home with that player unless both players have deleted the shared home.

### PR DESCRIPTION
**Pull request by Gobsmack90**

### CODE REVIEW

File: app/commands/home.js [hunk 1]
```diff
@@ -105,4 +126,100 @@ module.exports = Base => class extends Base {
...
+
+        let requestingPlayer = this.shareRequests[playerName][0];
+        let requestingAltName = this.shareRequests[playerName][1];
+
+        if (requestingPlayer) {
+            // Get Position & Rotation of accepting player, this is where the shared home will be located.
+            let position = await this.getPlayerPosition(playerName);
+            let rotation = await this.getPlayerRotation(playerName);
+            let world = await this.getPlayerDimension(playerName);
+
```
Comment (thorbaert) [inline app/commands/home.js:218]: We should do a check to ensure they aren't too far apart

File: app/commands/xp.js [hunk 2]
```diff
@@ -341,8 +341,8 @@ module.exports = Base => class extends Base {
             this.tellPlayer(playerName, `No pending experience point offers.`, 'red');
             return;
         }
-        let offeringPlayer = this.giveOffers[playerName][0];
-    
+
+        let offeringPlayer = this.giveOffers[playerName][0];
```
Comment (thorbaert) [inline app/commands/xp.js:345]: Why is this edited?

File: app/data/index.js [hunk 3]
```diff
@@ -43,6 +43,10 @@ module.exports = Base => class extends Base {
         return new Promise((resolve, reject) => {
             Player.findOne({ name: playerName }, (err, player) => {
                 if (err) return reject(err);
+                
+                if (player.shareHomes && player.shareHomes.hasOwnProperty(homeName)) {
```
Comment (thorbaert) [inline app/data/index.js:47]: This is an anti pattern and shouldn't be happening here. We may need to have a discussion about design pattern on this

**Review by thorbaert — CHANGES_REQUESTED** [review]

File: app/commands/xp.js (hunk 2 above)
Comment (Gobsmack90) [reply app/commands/xp.js:345]: i was looking through to see what i'd done in the past here and decided i didn't like the way stuff was spaced

**Review by thorbaert — APPROVED** [review]

**Pull request merged**